autopilot/autopilot.go

```diff
@@ -84,55 +89,44 @@ func (ap *Autopilot) SetConfig(c Config) error {
 	return ap.store.SetConfig(c)
 }
 
-// Actions returns the autopilot actions that have occurred since the given time.
-func (ap *Autopilot) Actions(since time.Time, max int) []Action {
-	panic("unimplemented")
-}
-
 func (ap *Autopilot) Run() error {
 	if err := ap.load(); err != nil {
 		return err
 	}
 
-	go ap.contractLoop()
-	go ap.hostScanLoop()
-	<-ap.stopChan
-	return nil // TODO
+	for {
+		select {
+		case <-ap.stopChan:
+			return nil
+		case <-ap.ticker.C:
+		}
+
+		// run host scan in separate goroutine, if a host scan is already
+		// running this is a no-op
+		go ap.s.tryPerformHostScan()
+
+		_ = ap.c.performContractMaintenance()
```

Comment:
This is mostly just an example of how the contract maintenance loop can be blocking, while the host scan is launched in a goroutine. I personally like the `tryPerformX` thing which is a no-op if the subsystem decides it should not run for whatever reason (e.g. like a scan already running or not enough time passing). Open for feedback though